### DISCUSSION TITLE
Licensing of derived code

### DISCUSSION THREAD

**Discussion by amalon** [openscenegraph/OpenSceneGraph#1186] [Q&A]
Hi, I'm trying to get my head around the OSG public license. Can anybody informally advise on whether it is compatible with the LGPL v2.1, in the sense of allowing code from OSG (e.g. an intersector class in osgUtil) to be used as a basis of a new intersector class in an LGPL v2.1 or GPLv2 licensed project, or would I have to keep the wxWindows exceptions intact in those specific source files? Can the resulting binary or library be distributed under LGPL/GPL or would those exceptions have to apply to the whole thing?

**Comment by robertosfield** [comment]
OSGPL is LGPL but with the WxWidgets relaxation.  If you reuse the code directly then it'll need to retain the original copyright notice.

Personally I'm OK with it being used in LGPL/GPL but I'm not the only author of the intersection code so can't waive rights of others.

On the code side I'm surprised that one would find value in the intersection class, while functional for the purposes that OSG requires, it's not state of art in any way.  I would have thought there are better dedicated intersection codes available that could be used as a better base.
